### PR TITLE
macOS Support

### PR DESCRIPTION
**Pull request by mominul**
This is a **super early** proof of concept, but I can now build OpenBangla Keyboard on macOS. There are many things left to do, but it is a good start.

Anyone can compile by following these steps:
```bash
brew update
brew install qt5
export CMAKE_PREFIX_PATH=$(brew --prefix qt5)
mkdir build
cd build
cmake .. -DCMAKE_INSTALL_PREFIX='/usr'
```
I am using macOS Sierra & Xcode 8.2

Here's the things has been done and needs to be done:
- [X] Compile `frontend`
- [X] Teach cmake how to compile swift files
- [X] Start working on `engine`

Valuable suggestions and discussions are welcome!

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,25 @@
...
 dist: trusty
-compiler:
-  - g++
-  - clang
 before_install:
-  - sudo add-apt-repository --yes ppa:beineri/opt-qt551-trusty
-  - sudo apt-get update -qq
-  - sudo apt-get install libibus-1.0-dev qt55base qt55declarative
-  - source /opt/qt55/bin/qt55-env.sh
+- if [ "$TRAVIS_OS_NAME" == "linux" ]; then
```
Comment (safwanrahman) [inline .travis.yml]: Better to have it in a script and run the script from travis ci. Then its much cleaner

Comment (mominul) [reply .travis.yml]: Thanks for the tip! I will try it next time when I work on this PR again.

**Comment by mominul** [comment]
Will have to start the work anew, so closing for now!

**Comment by marufmax** [comment]
any update? pleaseeee bring this to Mac! I love using it on my Arch maching.It would be awesome if I could use it on my Mac too

**Comment by mominul** [comment]
@marufmax The work has been suspended for lack of a Mac to develop it with. But I have a Mac now, so I intend to start the work again!

**Comment by ASHspace** [comment]
@mominul any update for the Mac version?

**Comment by mominul** [comment]
@ASHspace I intend to start working on the Mac version after I release the 3.0.0.

**Comment by ASHspace** [comment]
@mominul thanks, will for the mac version in future.

**Pull request closed without merging**